FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,47 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		let context = self.context
+		context.perform {
```

Comment:
During the course this way of having a local reference to the context to avoid capturing self inside of the `context.perform` block is taught.
The same can be achieved using capture list.
It saves one line and one variable.

No need to make this amendment, this is just for your interest.

```suggestion
		context.perform { [context] in
```

Reply:
Would this be an alternative way to write it?


```
context.perform {
   [weak self] in
  guard let context = self?.context else { return }
  ....
```

Reply:
It is not 100% the same. In your example commands after `guard ... let` will not be executed in case `self` is deallocated whereas in `context.perform { [context] in ... }` they will as it has no check for `self` deallocation

Reply:
You're right. It's a using strong reference to avoid a weak self.  I changed it to { [context] in ... }  as suggested.  @danillahtin